drivers/serial/uart_mcux.c

```diff
@@ -422,7 +424,7 @@ static const struct uart_mcux_config uart_mcux_##n##_config = {		\
 #define UART_MCUX_INIT(n)						\
 	PINCTRL_DEFINE(n)						\
 									\
-	static struct uart_mcux_data uart_mcux_##n##_data = {		\
+	static struct uart_mcux_data UART_DT_INST_DATA_NAME(n) = {	\
```

Comment:
I like the idea of providing macros for data/config entries, but these could be generic to the device level, they don't seem to have any subsystem specifics. Example:

```c
#define DEVICE_DT_INST_DATA(n, type) static type UTIL_CAT(Z_DEVICE_DT_DEV_NAME(DT_DRV_INST(n)), _data)

DEVICE_DT_INST_DATA(n, struct uart_mcux_data) = { ... };
```

Reply:
I wanted the 'API' name as part of the defines.  We can implement them in terms of a common `DEVICE_DT_INST_DATA`.

Reply:
(not a request for change, just a thought) I would even go a step further and standardize everything :)

You could use the `DT_DRV_COMPAT` to do some standard things. This would be similar to how zTest requires the fixture struct name to be the `SUITE_NAME##_fixture`. In that sense you could standardize the data/config name macros to always work the same `UTIL_CAT(UTIL_CAT(DT_DRV_COMPAT, _data), n)` or some variation thereof.